### ISSUE TITLE
Pylint does not report undefined variable in if..else branch

### ISSUE THREAD

**Issue by iulian-birlica** [PyCQA/pylint#7462]
### Bug description

```python
# pylint: disable=missing-docstring
import random
def function():
    if random.randrange(2) == 0:
        test = "something"
        print(test)
    else:
        print(test) # pylint should complain
```

### Command used

```shell
pylint test.py
```

### Pylint output

```shell
------------------------------------
Your code has been rated at 10.00/10
```

### Expected behavior

Expected some kind of undefined variable warning.

### Pylint version

```shell
pylint 2.15.2
astroid 2.12.9
Python 3.9.2 (default, Dec 16 2021, 12:56:11)
[GCC 6.3.0 20170516]
```

### OS / Environment

No LSB modules are available.
Distributor ID:	Debian
Description:	Debian GNU/Linux 9.13 (stretch)
Release:	9.13
Codename:	stretch

**Comment by jacobtylerwalls** [comment]
Thanks for the report. `undefined-variable` and `used-before-assignment` are almost synonymous. Duplicate of #1727
